azure/scope/cluster.go

```diff
@@ -230,6 +230,40 @@ func (s *ClusterScope) PublicIPSpecs() []azure.ResourceSpecGetter {
 	return publicIPSpecs
 }
 
+// SetAPIServerLBBackendPoolName sets the name of the backend pool for apiserver LB.
+func (s *ClusterScope) SetAPIServerLBBackendPoolName() {
```

Comment:
this looks good, I think my only comment/question is whether it would be possible to move these functions to the default webhook file and use them in both the webhook and in `SetLBBackendPoolName()` below so we're sharing the default logic with the webhook in case anything changes later.